nixos/modules/virtualisation/containers-next/default.nix

```diff
@@ -0,0 +1,571 @@
+            <prompt># </prompt>ip6tables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+            </screen>
+          '';
+        };
+      } // (optionalAttrs (v == 4) {
+        nat = mkOption {
+          default = true;
+          type = types.bool;
+          description = ''
+            Whether to set-up a basic NAT to enable internet access for the nspawn containers.
+          '';
+        };
+      });
+    in
+      assert elem type [ "veth" "zone" ]; {
+        v4 = mkIPOptions 4;
+        v6 = mkIPOptions 6;
+      } // optionalAttrs (type == "zone") {
+        hostAddresses = mkOption {
+          default = [];
+          type = types.listOf types.str;
+          description = ''
+            Address of the container on the host-side, i.e. the
+            subnet and address assigned to <literal>vz-&lt;name&gt;</literal>.
+          '';
+        };
+      };
+
+  mkImage = name: config:
+    { container = import "${toString config.nixpkgs}/nixos/lib/eval-config.nix" {
+        system = pkgs.stdenv.hostPlatform.system;
+        modules = [
+          ./container-profile.nix
+          ({ pkgs, ... }: {
+            networking.hostName = name;
+            systemd.network.networks."20-host0" = {
+              address = mkIf (cfg.${name}.network != null) (
+                cfg.${name}.network.v4.static.containerPool
+                ++ cfg.${name}.network.v6.static.containerPool
+              );
+            };
+          })
+        ] ++ (config.system-config);
+        prefix = [ "nixos" "containers" "instances" name "config" ];
+      };
+      inherit config;
+    };
+
+  mkContainer = cfg: let inherit (cfg) container config; in mkMerge [
+    {
+      execConfig = mkMerge [
+        {
+          Boot = false;
+          Parameters = "${container.config.system.build.toplevel}/init";
+          Ephemeral = yesNo config.ephemeral;
+          KillSignal = "SIGRTMIN+3";
+          X-ActivationStrategy = config.activation.strategy;
+          PrivateUsers = mkDefault "pick";
+        }
+        (mkIf (!config.ephemeral) {
+          LinkJournal = mkDefault "guest";
+        })
+      ];
+      filesConfig = mkMerge [
+        { PrivateUsersChown = mkDefault "yes"; }
+        (mkIf config.sharedNix {
+          BindReadOnly = [ "/nix/store" "/nix/var/nix/db" "/nix/var/nix/daemon-socket" ];
+        })
+      ];
+      networkConfig = mkMerge [
+        (mkIf (config.zone != null || config.network != null) {
+          Private = true;
+          VirtualEthernet = "yes";
+        })
+        (mkIf (config.zone != null) {
+          Zone = config.zone;
+        })
+        (mkIf (config.forwardPorts != []) {
+          Port = config.forwardPorts;
+        })
+      ];
+    }
+    (mkIf (!config.sharedNix) {
+      extraDrvConfig = let
+        info = pkgs.closureInfo {
+          rootPaths = [ container.config.system.build.toplevel ];
+        };
+      in pkgs.runCommand "bindmounts.nspawn" { }
+        ''
+          touch $out
```

Comment:
```suggestion
```
file is already created with the redirect below or is there anything different about this?